cli/src/query.rs

```diff
@@ -24,10 +33,94 @@ pub struct QueryCommand {
     #[arg(long)]
     pub value: bool,
 
+    /// Export the value and all metadata of selected field in the specified format.
+    ///
+    /// This flag cannot be used along with the following flags: --doc, --contract, --type, --default, --value
+    #[arg(long, short, value_enum, default_value_t, conflicts_with_all(["doc", "contract", "typ", "default", "value"]))]
+    pub format: MetadataExportFormat,
+
+    /// Output file. Standard output by default
+    #[arg(short, long, conflicts_with_all(["doc", "contract", "typ", "default", "value"]))]
+    pub output: Option<PathBuf>,
+
     #[command(flatten)]
     pub inputs: InputOptions<ExtractFieldOnly>,
 }
 
+#[derive(Clone, Debug, Serialize)]
+struct QueryResult {
+    pub doc: Option<String>,
+    pub typ: Option<LabeledType>,
+    pub contracts: Vec<LabeledType>,
+    pub optional: bool,
+    pub not_exported: bool,
+    pub priority: MergePriority,
+    pub sub_fields: Option<Vec<Ident>>,
+    pub value: Option<RichTerm>,
+}
+
+impl From<Field> for QueryResult {
+    fn from(field: Field) -> Self {
+        let sub_fields = match field.value {
+            Some(ref val) => match val.as_ref() {
+                Term::Record(record) if !record.fields.is_empty() => {
+                    let mut fields: Vec<_> = record.fields.keys().collect();
+                    fields.sort();
+                    Some(fields.into_iter().map(LocIdent::ident).collect())
+                }
+                Term::RecRecord(record, dyn_fields, ..) if !record.fields.is_empty() => {
+                    let mut fields: Vec<_> = record.fields.keys().map(LocIdent::ident).collect();
+                    fields.sort();
+                    let dynamic = Ident::from("<dynamic>");
+                    fields.extend(dyn_fields.iter().map(|_| dynamic));
+                    Some(fields)
+                }
+                // Empty record has empty sub_fields
+                Term::Record(..) | Term::RecRecord(..) => Some(Vec::new()),
+                // Non-record has no concept of sub-field
+                _ => None,
+            },
+            None => None,
+        };
+
+        QueryResult {
+            doc: field.metadata.doc,
+            typ: field.metadata.annotation.typ,
+            contracts: field.metadata.annotation.contracts,
+            optional: field.metadata.opt,
+            not_exported: field.metadata.not_exported,
+            priority: field.metadata.priority,
+            sub_fields,
+            value: field.value,
+        }
+    }
+}
+
+impl QueryResult {
+    pub fn filter_for_export(self) -> serde_json::Map<String, serde_json::Value> {
+        let mut query_res = serde_json::to_value(&self).unwrap();
+        let query_res = query_res.as_object_mut().unwrap();
+
+        if self.value.is_none() {
```

Comment:
I looked it up and it seems that you can achieve the same a bit more declaratively by slapping the following serde attribute on the corresponding field of the enum:

```rust
struct QueryResult {
  #[serde(skip_serializing_if = "Option::is_none")]
  value: Option<RichTerm>,
  // etc..
}
```

And get rid of the `if`s here.